### PR TITLE
fix opensearch and kinesis test markers

### PR DESCRIPTION
**Pull request by alexrashed**
## Motivation
While working on #11133, I saw that the test module contains a few tests with "needs fixing".
Also, @bentsku made me aware of the fact that the opensearch test module contains a lot of `markers.aws.unkown` which we try to get rid of with lots of PRs in the last few months. Just adding a few cross-links to connect these efforts a bit:
- https://github.com/localstack/localstack/pull/10994
- https://github.com/localstack/localstack/pull/10991
- https://github.com/localstack/localstack/pull/10987
- https://github.com/localstack/localstack/pull/10929
- https://github.com/localstack/localstack/pull/10920
- https://github.com/localstack/localstack/pull/10892
- https://github.com/localstack/localstack/pull/10653
- https://github.com/localstack/localstack/pull/9151
- https://github.com/localstack/localstack/pull/9141

## Changes
- Fixes all but one non-AWS validated kinesis test (the last one fails with an Internal Server Error / 500 by AWS).
- Basically marks most of the opensearch tests as `needs_fixing` (the domain creation fixture is not working against AWS yet), and some as `only_localstack` (there are a lot of tests for the different endpoint strategies).

### CODE REVIEW

**Review by bentsku — APPROVED** [review]
LGTM! Thanks a lot for taking care of this, this helps us a lot 🙏 also, nice to see lots of Kinesis tests validated now! Really love the `aws_http_client_factory` fixture ❤️
